### PR TITLE
feat(codewhisperer): enable UTG (Unit test generator) for java/python

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Solution

* Enable UTG (unit test generator) feature flag for CrossFile userGroup
* Make changes to ensure supplemental context telemetry being sent regardless the userGroup it belongs (we need `metadata.codewhispererSupplementalContextIsUtg` field to differentiate UTG feature acceptance rate vs. CrossFile feature)

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/util/supplementalContext/utgUtils.ts [hunk 1]
```diff
@@ -51,6 +53,10 @@ export async function fetchSupplementalContextForTest(
         return undefined
     }
 
+    if (CodeWhispererUserGroupSettings.instance.userGroup !== UserGroup.Utg) {
+        return []
+    }
```
Comment (Will-ShaoHua) [inline src/codewhisperer/util/supplementalContext/utgUtils.ts:58]: Not returning `undefined` on purpose, we need supplemental context relevant fields sent no matter what userGroup the user belongs.

File: .changes/next-release/Feature-efc3928c-3e15-4e06-96e0-13d10c3496ab.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "CodeWhisperer: Improve file context fetching logic"
```
[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply .changes/next-release/Feature-efc3928c-3e15-4e06-96e0-13d10c3496ab.json:3]: yea we aligned not exposing too much detailed info to the customers.

File: src/codewhisperer/util/userGroupUtil.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ export class CodeWhispererUserGroupSettings {
 
     private guessUserGroup(): UserGroup {
         const randomNum = Math.random()
-        const result = randomNum <= 1 / 2 ? UserGroup.Control : UserGroup.CrossFile
+        const result = randomNum <= 1 / 2 ? UserGroup.Control : UserGroup.Utg
```
Comment (leigaol) [inline src/codewhisperer/util/userGroupUtil.ts:55]: This has a conflict with another PR of yours.

Comment (Will-ShaoHua) [reply src/codewhisperer/util/userGroupUtil.ts:55]: yes, the other one is to remove cross file from the A/B
and this one it to add utg to the A/B
i know it's a bit confusing, but I do this for seperate these 2feats in 2 PRs

**Comment by leigaol** [comment]
Could you fix the unit tests? Thank you!

**Review by leigaol — APPROVED** [review]

File: .changes/next-release/Feature-efc3928c-3e15-4e06-96e0-13d10c3496ab.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply .changes/next-release/Feature-efc3928c-3e15-4e06-96e0-13d10c3496ab.json:3]: @JadenSimon does this look better 
`CodeWhisperer: Improve suggestion quality with enhanced file context fetching`

[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply .changes/next-release/Feature-efc3928c-3e15-4e06-96e0-13d10c3496ab.json:3]: however the suggestion quality  is not guaranteed.. plus we have a A/B testing

[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply .changes/next-release/Feature-efc3928c-3e15-4e06-96e0-13d10c3496ab.json:3]: or just `Enhance file context fetching`

[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply .changes/next-release/Feature-efc3928c-3e15-4e06-96e0-13d10c3496ab.json:3]:
```suggestion
	"description": "CodeWhisperer: Improve suggestion quality with enhanced file context fetching"
```

**Review by JadenSimon — APPROVED** [review]

**Pull request merged**